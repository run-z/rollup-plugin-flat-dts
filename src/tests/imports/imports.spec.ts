import { describe, expect, it } from '@jest/globals';
import { testDts } from '../test-dts';

describe('For imports with `.js` extension', () => {
  it('emits type definitions', async () => {
    const {
      files: [{ content }],
    } = await testDts('imports', {
      entries: { imported: {} },
    });

    expect(content).toMatchSnapshot();
  });
  it('emits declaration map', async () => {
    const { files } = await testDts('imports', {
      entries: { imported: {} },
      compilerOptions: { declarationMap: true },
    });

    expect(files).toHaveLength(2);

    const [dts, sourceMap] = files;

    expect(dts.path).toBe('index.d.ts');
    expect(sourceMap.path).toBe('index.d.ts.map');

    expect(dts.content).toMatchSnapshot('dts');
    expect(sourceMap.content).toMatchSnapshot('source map');
  });
});
