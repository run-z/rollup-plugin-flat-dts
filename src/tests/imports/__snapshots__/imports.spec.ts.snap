// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`For imports with \`.js\` extension emits declaration map: dts 1`] = `
"declare module "test-package/imported" {
    export interface Imported {
        readonly name: 'imported';
    }
}

declare module "test-package" {
    export * from "test-package/imported";
}
//# sourceMappingURL=index.d.ts.map
"
`;

exports[`For imports with \`.js\` extension emits declaration map: source map 1`] = `"{"version":3,"sources":["src/tests/imports/imported.ts","src/tests/imports/importer.ts"],"names":[],"mappings":";IAAA,MAAM,WAAW,QAAQ;QACvB,QAAQ,CAAC,IAAI,EAAE,UAAU,CAAC;KAC3B;;;;ICFD,sCAA8B","file":"index.d.ts","sourceRoot":""}"`;

exports[`For imports with \`.js\` extension emits type definitions 1`] = `
"declare module "test-package/imported" {
    export interface Imported {
        readonly name: 'imported';
    }
}

declare module "test-package" {
    export * from "test-package/imported";
}
"
`;
